pkg/controller/inferenceservice/resources/istio/virtualservice.go

```diff
@@ -271,8 +314,14 @@ func (r *VirtualServiceBuilder) CreateVirtualService(isvc *v1alpha2.InferenceSer
 		}
 		httpRoutes = append(httpRoutes, explainRoute)
 	}
+
 	// extract the virtual service hostname from the predictor hostname
-	serviceURL := fmt.Sprintf("%s://%s%s", "http", serviceHostname, constants.InferenceServicePrefix(isvc.Name))
+	var serviceURL string
+	if _, ok := annotations["custom.urls"]; ok {
```

Comment:
Can we make `ServiceCustomURL` function more generic so that it handles both custom and standard urls 
`getServiceURL(urlPaths, serviceHostName)`

Reply:
accept